app/scripts/controllers/transactions/index.js

```diff
@@ -337,9 +337,19 @@ export default class TransactionController extends EventEmitter {
    *
    * @param txParams
    * @param origin
+   * @param transactionType
    * @returns {txMeta}
    */
-  async addUnapprovedTransaction(txParams, origin) {
+  async addUnapprovedTransaction(txParams, origin, transactionType) {
+    if (
```

Comment:
You could do this instead:
```
const allowedTransactionTypes = [TRANSACTION_TYPES.SWAP, TRANSACTION_TYPES.SWAP_APPROVAL];
if (transactionType !== undefined && !allowedTransactionTypes.includes(transactionType)) {
throw new Error(
    `TransactionController - invalid transactionType value: ${transactionType}`,
  );
}
```

Reply:
Just updated

Reply:
total nit but `allowedTransactionTypes` feels a bit misleading in the context of this method. Maybe `allowedTransactionTypeParam`. I don't know?

Reply:
I added a new SWAP_TRANSACTION_TYPES constant.